### DISCUSSION TITLE
why doesn't my file show up in coverage unless I write a test for it?

### DISCUSSION THREAD

**Discussion by mildfuzz** [vitest-dev/vitest#3202] [Q&A] [answered]
I have any issue, where people are able to add code and so long as they don't add tests for their modules, the reporter values do not change. I need it to be able to scan all files and report untested branch logic.

```js
const config = {
  plugins: [react()],
  resolve: {
    alias,
  },
  test: {
    coverage: {
      provider: 'istanbul',
      reporter: ['cobertura', 'json', 'html', 'text-summary'],
      reportsDirectory: './coverage',
      branches: 100,
      src: process.cwd() + '/src',
    },
    setupFiles: './setupTests.js',
    exclude: [...configDefaults.exclude, './.next/'],
    environment: 'happy-dom',
  },
}
```

**Comment by sheremet-va** [comment, marked as answer]
Did you try `test.coverage.all: true`?

**Reply by mildfuzz** [reply]
Okay, so this has improved the situation. I have had to refine the `include` and `exclude` but now it includes files that are explicitly marked with `/* istanbul ignore file */`

**Reply by sheremet-va** [reply]
You probably need `-- @preserve`: https://vitest.dev/guide/coverage.html#ignoring-code

**Reply by mildfuzz** [reply]
Thanks

**Reply by mfa-leanix** [reply]
@sheremet-va, setting `coverage.all` doesn't work for me. I am using Vitest `3.1.3`.
